src/components/Comment/index.spec.tsx

```diff
@@ -1,82 +1,309 @@
+    return { store };
+  };
 
-    expect(root.find(`.${styles.form}`)).toHaveLength(1);
-    expect(root.find(`.${styles.textarea}`)).toHaveLength(1);
+  const createFakeChangeEvent = ({
+    name,
+    value,
+  }: {
+    name: string;
+    value: string;
+  }) => {
+    return createFakeEvent({ target: { name, value } });
+  };
+
+  const createFakeTextareaRef = (currentProps = {}) => {
+    return {
+      ...React.createRef(),
+      current: {
+        ...document.createElement('textarea'),
+        addEventListener: jest.fn(),
+        removeEventListener: jest.fn(),
+        focus: jest.fn(),
+        ...currentProps,
+      },
+    };
+  };
+
+  it('requires a comment when readOnly=true', () => {
+    expect(() => render({ readOnly: true, commentId: null })).toThrow(
+      /initialComment for commentId=null cannot be empty/,
+    );
+  });
+
+  it('throws when a commentId is not mapped to a comment', () => {
+    expect(() => render({ readOnly: false, commentId: 1 })).toThrow(
+      /No comment was mapped/,
+    );
   });
 
-  it('renders a comment when provided', () => {
-    const comment = createFakeComment();
+  it('renders a comment when readOnly=true', () => {
+    const content = 'Example of a comment';
+    const comment = createFakeExternalComment({ comment: content });
+    const { store } = dispatchComment({ comment });
 
-    const root = render({ comment });
+    const root = render({ commentId: comment.id, store, readOnly: true });
 
     expect(root.find(`.${styles.form}`)).toHaveLength(0);
     expect(root.find(`.${styles.comment}`)).toHaveLength(1);
-    expect(root.find(`.${styles.comment}`).html()).toContain(comment.content);
+    expect(root.find(`.${styles.comment}`).html()).toContain(content);
     expect(root.find(FontAwesomeIcon)).toHaveLength(1);
   });
 
   it('sanitizes the content of a comment', () => {
-    const comment = createFakeComment({ content: '<span>foo</span>' });
+    const comment = createFakeExternalComment({ comment: '<span>foo</span>' });
+    const { store } = dispatchComment({ comment });
 
-    const root = render({ comment });
+    const root = render({ commentId: comment.id, store, readOnly: true });
 
     expect(root.find(`.${styles.comment}`).html()).not.toContain(
-      comment.content,
+      comment.comment,
     );
     // HTML `span` are removed.
     expect(root.find(`.${styles.comment}`).html()).toContain('foo');
   });
 
-  it('renders a form to edit a comment when the readOnly prop is false', () => {
-    const comment = createFakeComment();
-    const root = render({ comment, readOnly: false });
+  it('renders a form to edit a comment when readOnly=false', () => {
+    const comment = createFakeExternalComment({ comment: 'Example' });
+    const { store } = dispatchComment({ comment });
+    const root = render({ commentId: comment.id, store, readOnly: false });
 
     expect(root.find(`.${styles.form}`)).toHaveLength(1);
```

Comment:
In the test for `renders a comment when readOnly=true`, you assert both that `styles.comment` is present and that `styles.form` is not, yet in this test you only assert that `styles.form` is present. Should there be consistency between the tests?

Reply:
whoops, this one needs an assertion that the comment is not present.